CodeEditModules/Modules/WorkspaceClient/src/Live.swift

```diff
@@ -128,78 +126,38 @@ public extension WorkspaceClient {
-
-                source.setEventHandler {
-                    // Something has changed inside the directory
-                    // We should reload the files.
-                    guard !isRunning else { // this runs when a file change is detected but is already running
-                        anotherInstanceRan += 1
-                        return
-                    }
-                    isRunning = true
-                    flattenedFileItems = [workspaceItem.id: workspaceItem]
-                    _ = try? rebuildFiles(fromItem: workspaceItem)
-                    while anotherInstanceRan > 0 { // TODO: optimise
-                        let somethingChanged = try? rebuildFiles(fromItem: workspaceItem)
-                        anotherInstanceRan = !(somethingChanged ?? false) ? 0 : anotherInstanceRan - 1
-                    }
-                    subject.send(workspaceItem.children ?? [])
-                    startListeningToDirectory()
-                    isRunning = false
-                    anotherInstanceRan = 0
-                    // reload data in outline view controller through the main thread
-                    DispatchQueue.main.async { onRefresh() }
-                }
-
-                source.setCancelHandler {
-                    close(descriptor)
-                }
-
-                source.resume()
-
-                sources[item.id] = source
+        FileItem.watcherCode = {
+            // Something has changed inside the directory
+            // We should reload the files.
+            guard !isRunning else { // this runs when a file change is detected but is already running
+                anotherInstanceRan += 1
+                return
             }
-
-            // test for deleted directories and remove their listeners
-            for (id, _) in sources {
-                var childExists = false
-                for item in flattenedFileItems.values {
-                    childExists = item.id == id ? true : childExists
-                }
-
-                if !childExists {
-                    sources.removeValue(forKey: id)?.cancel()
-                }
+            isRunning = true
+            flattenedFileItems = [workspaceItem.id: workspaceItem]
+            _ = try? rebuildFiles(fromItem: workspaceItem)
+            while anotherInstanceRan > 0 { // TODO: optimise
+                let somethingChanged = try? rebuildFiles(fromItem: workspaceItem)
+                anotherInstanceRan = !(somethingChanged ?? false) ? 0 : anotherInstanceRan - 1
             }
+            subject.send(workspaceItem.children ?? [])
+            startListeningToDirectory()
+            isRunning = false
+            anotherInstanceRan = 0
+            // reload data in outline view controller through the main thread
+            DispatchQueue.main.async { onRefresh() }
         }
 
+        // no longer functional. Directory listening code has been moved to the FileItem.
+        func startListeningToDirectory() {}
```

Comment:
If this method is empty and generally does nothing, may we remove it?

Reply:
done